DOAJ/doaj
# improve ur/new_app separation script ready for re-run

# Improve Update Request/New Application script

*Please don't delete any sections when completing this PR template; instead enter **N/A** for checkboxes or sections which are not applicable, unless otherwise stated below*

See https://github.com/DOAJ/doajPM/issues/3362

In order to correct a bug introduced via a previous PR which incorrectly classified new applications as update requests we have re-engineered the original script to separate URs from Applications via an `application_type` field.  This script encompasses all rules that allow us to distinguish between these two types of application, and can be re-run as many times as needed.

It follows the algorithm here
![URAppSplitWorkflow](https://user-images.githubusercontent.com/856138/190507579-14befdd9-4f48-40d7-bc0f-f139e71598f0.png)


## Categorisation

This PR...

## Basic PR Checklist

- [x] **N/A** FeatureMap annotations have been added
- [x] Unit tests have been added/modified - test is a post migration test script
- [x] **N/A** Functional tests have been added/modified
- [x] Code has been run manually in development, and functional tests followed locally
- [x] No deprecated methods are used
- [x] No magic strings/numbers - all strings are in `constants` or `messages` files
- [x] ES queries are wrapped in a Query object rather than inlined in the code
- [x] Where possible our common library functions have been used (e.g. dates manipulated via `dates`)
- [x] If needed, migration has been created and tested locally
- [ ] Release sheet has been created, and completed as far as is possible https://docs.google.com/spreadsheets/d/1Bqx23J1MwXzjrmAygbqlU3YHxN1Wf7zkkRv14eTVLZQ/edit
- [x] **N/A** Documentation updates - if needed - have been identified and prepared for inclusion into main documentation (e.g. added and highlighted/commented as appropriate to this PR)
    - [x] Core model documentation: https://docs.google.com/spreadsheets/d/1lun2S9vwGbyfy3WjIjgXBm05D-3wWDZ4bp8xiIYfImM/edit
    - [x] Events and consumers documentation: https://docs.google.com/spreadsheets/d/1oIeG5vg-blm2MZCE-7YhwulUlSz6TOUeY8jAftdP9JE/edit
- [x] There has been a recent merge up from `develop` (or other base branch)

## Testing

This will need to be tested on the test server with a fresh anon_export from live.

Recommend that we run the migration as follows:

1. Run the script `portality/migrate/20220915_3362_ur_relabelling/post_migration.py` - this should tell us to what degree contents of the index are mis-classified
2. Run the migration `python portality/upgrade.py -u portality/migrate/20220915_3362_ur_relabelling/migrate.json`
3. Re-run the post-migration script above - this should confirm that the migration has been successful

The DOAJ team should then check that their issue as reported has been resolved.  They should also spot check a number of other applications and update requests to confirm that they are correctly classified. The post-migration will provide a sample of example IDs for each category of the migration.

## Deployment

What deployment considerations are there? (delete any sections you don't need)

### Configuration changes

** N/A **

### Scripts

** N/A **

### Migrations

Run the migration `python portality/upgrade.py -u portality/migrate/20220915_3362_ur_relabelling/migrate.json`

Afterwards, worth running `portality/migrate/20220915_3362_ur_relabelling/post_migration.py` to confirm success

### Monitoring

** N/A **

### New Infrastructure

** N/A **

### Continuous Integration

** N/A **
